plugins/alexa/README.md

```diff
@@ -0,0 +1,151 @@
+
+# Requirements
+```
+sudo pip3 install cherrypy
+sudo pip3 install simplejson
+```
+
+# Testing
+https://echosim.io/
+
+# Configuration
+
+## plugin.conf
+basic configuration
+```
+[alexa]
+    class_name = Alexa
+    class_path = plugins.alexa
+```
+
+you may change host/ip and port of the web-service
+```
+[alexa]
+    class_name = Alexa
+    class_path = plugins.alexa
+    service_host = "0.0.0.0"
+    service_port = 9000
+```
+
+## items.conf
+implemented actions:
+- turnOn
+- turnOff
+- setTargetTemperature
+- incrementTargetTemperature
+- decrementTargetTemperature
+- setPercentage
+- incrementPercentage
+- decrementPercentage
+
+specify supported actions space-separated
+```
+[item]
+type = foo
+alexa_name = "Diningroom Lamp"
+alexa_actions = turnOn turnOff
+```
+
+you may omit the `alexa_name`, it will use the item's `name`
+```
+[item]
+type = foo
+name = "Diningroom Lamp"
+alexa_actions = turnOn turnOff
+```
+
+you can use multiple items for specific actions using the same alexa-name.
+```
+[item_only_on]
+type = foo
+alexa_name = "Diningroom Lamp"
+alexa_actions = turnOn
+
+[item_only_off]
+type = foo
+alexa_name = "Diningroom Lamp"
+alexa_actions = turnOff
+```
+
+the device-identifier is automatically deduced from the `alexa_name` - but you can specify it explicitly using `alexa_device`
+```
+[[item_only_on]]
+type = foo
+alexa_name = "Diningroom Lamp"
+alexa_actions = turnOn
+
+[[item_only_off]]
+type = foo
+alexa_name = "Diningroom Lamp"
+alexa_actions = turnOff
+```
+
+alexa supports "friendly descriptions", you can set it using `alexa_description`
+```
+[item]
+type = foo
+alexa_name = "Diningroom Lamp"
+alexa_actions = turnOn turnOff
+alexa_description = "The pompous dining room lamp in the west-wing"
+```
+
+you can define `alexa_name` & `alexa_description` centrally in one item and reference the device in other items only by using the `alexa_device` (you must always define a type though!)
+```
+[root]
+  [[livingroom_lamps]]
+  type = foo
+  alexa_device = livingroom_lamps
+  alexa_name = "Livingroom"
+  alexa_description = "Couch and Main Livingroom-Lamps"
+
+    [[[couch]]]
+    type = bool
+    alexa_device = livingroom_lamps
+    alexa_actions = "turnOn turnOff"
+    knx_dpt = 1
+		knx_listen = 1/2/1
+    knx_init = 1/2/1
+    knx_send = 1/2/0
+
+    [[[main]]]
+    type = bool
+    alexa_device = livingroom_lamps
+    alexa_actions = "turnOn turnOff"
+    knx_dpt = 1
+		knx_listen = 1/2/11
+    knx_init = 1/2/11
+    knx_send = 1/2/10
+```
+
+## logging.yaml
+you can enable debug logging for the alexa-plugin specifically:
+```
+loggers:
+  plugins.alexa:
+    level: DEBUG
+root:
+    level: INFO
+    handlers: [file, console]
+```
```

Comment:
Do we really want to document how to setup the logging for a plugin? This should be/is documented in a common section of SH.py documentation.